src/lib.rs

```diff
@@ -1,6 +1,29 @@
 use std::collections::HashMap;
 use std::collections::HashSet;
 use std::hash::Hash;
+use std::error::Error;
+use std::fmt;
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash, PartialOrd, Ord)]
+pub struct LengthError;
```

Comment:
We could make this type more useful by including the lengths of both predictions and ground truth. The implementation of `Display` could then be improved accordingly.

Reply:
Cleaned up, adds the two variables to the struct and to the display.